run/cli_ioproducer.go

```diff
@@ -0,0 +1,35 @@
+package run
+
+import (
+	"context"
+	"io"
+	"log"
+	"os"
+)
+
+// CliIOProducer is the IOProducer for the CliRunner. The input and output paths
+// are used to configure the input and output readers and writers. If the paths
+// are empty, os.Stdin and os.Stdout are used.
+func CliIOProducer(_ context.Context, cfg CLIRunnerConfig) IOData {
+	reader := os.Stdin
+	if cfg.Runner.Input.Path != "" {
+		r, err := os.Open(cfg.Runner.Input.Path)
```

Comment:
Whose responsible is it to close the file?

Reply:
The `Decoder`, good catch. I'll add the missing code.